doc/user/content/sql/drop-index.md

```diff
@@ -27,8 +30,8 @@ VIEW`](../create-view).
 
 ### Primary indexes
 
-By default, materialized views only have one index, which we call the "primary
-index," and stores the result set of the materialized view's embedded `SELECT`
+By default, materialized views only have one index, called the "primary
```

Comment:
The term "primary index" has a distinct meaning in the world of databases, and our "primary indexes" do not work like primary indexes do in traditional databases, hence the oddly phrased "which we call." I was trying to intimate at this fact, but perhaps wasn't explicit enough.

Reply:
@sploiselle - Can you give me a quick summary of the difference between Materialize primary indexes and trad db primary indexes? Basically, what user expectations will we confound?

Reply:
The term primary index in a traditional database comes freighted with meaning, especially for folks familiar with that database's behavior regarding indices. For example, the primary index is often thought of as the index that enforces the invariant that provides a table's primary key (i.e. the data in the columns in the primary is a unique set in each row, which also implies that it is not null). In some systems, the primary key also describes how to store the canonical version of the data on disk (or in memory)––i.e. all other indices on the table are derivatives of the primary index.

What we call our "primary index" is just the first index created on a view. You can drop it and, if you have another covering index on the view (covering in this context just means that the index includes all of the columns), I don't believe there are any untoward effects. However, if you drop the "primary index" and do not have another covering index (if it's even possible to have a non-covering index in this case?), you can no longer query the view (in the general case), or at least cannot query any column not included in an index. In a traditional RDBMS, if you drop the primary index, you can still query the table, it will just be very slow––i.e. in the general case, the RDBMS will essentially create a "new" primary index whose primary key is the empty set and each row is essentially just a bag of bytes you have to sift through "manually" to do any kind of sorting.

There are also things like this that are just very dramatically different from the kind of behavior one expects out of primary indices in a traditional database:

> You cannot exclude any columns from being in the index’s “value” set. For example, if your view embedded `SELECT a, b, c..`., all indexes will contain `{a, b, c}` as their values.

In a traditional RDBMS, you would only include the on-indexed columns in the value set.

All of this is "basically" true, but there are certainly edge cases that this glosses over. Indices in databases are an incredibly intricate topic with few universal truths because each RDBMS makes slightly different tradeoffs with how they handle indices to achieve some desirable set of performance characteristics.

Reply:
@sploiselle Thanks! Does this work?

> By default, materialized views only have one index, which we call the "primary
index," and which stores the result set of the materialized view's embedded `SELECT`
statement. You can identify the primary index by its name, which follows the
format `<view name>_primary_idx`.
>
> The materialized view primary index doesn't follow the constraints for typical for primary indexes in traditional databases; it is simply the first index created on a view. However, if the view doesn't have another index that covers all of the columns, you will not be able to query any columns not included in an existing index.

Reply:
The second paragraph needs some TLC copy-edit-wise, but I believe captures the gist. Ping me once you've cleaned it up.